src/Protocol/Smtp/Auth/Xoauth2.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Laminas\Mail\Protocol\Smtp\Auth;
+
+use Laminas\Mail\Protocol\Smtp;
+use Laminas\Mail\Protocol\Xoauth2\Xoauth2 as Xoauth2AuthEncoder;
+
+use function array_replace_recursive;
+use function is_array;
+
+/**
+ * Performs PLAIN authentication
+ */
+class Xoauth2 extends Smtp
+{
+    /**
+     * PLAIN username
+     *
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * PLAIN password
+     *
+     * @var string
+     */
+    protected $accessToken;
+
+    /**
+     * @param  string $host   (Default: 127.0.0.1)
+     * @param  int    $port   (Default: null)
+     * @param  array  $config Auth-specific parameters
+     */
+    public function __construct($host = '127.0.0.1', $port = null, $config = null)
```

Comment:
```suggestion
    public function __construct($host = '127.0.0.1', $port = null, ?array $config = null)
```
Signature should match parent's one